### PR TITLE
Review changes for #2911

### PR DESCRIPTION
**Pull request by coot**
- network-mux: miniProtocolStateMap
- StructLinger: added todo

### CODE REVIEW

File: ouroboros-network-framework/src/Ouroboros/Network/Linger.hsc [hunk 1]
```diff
@@ -14,6 +14,7 @@ import           Network.Socket (Socket)
 import qualified Network.Socket as Socket
 
 
+-- TODO: tobe removed once a new version of `network` library is released.
```
Comment (karknu) [inline ouroboros-network-framework/src/Ouroboros/Network/Linger.hsc:17]: to be

Comment (coot) [reply ouroboros-network-framework/src/Ouroboros/Network/Linger.hsc]: done

**Review by karknu — APPROVED** [review]
LGTM.

**Comment by coot** [comment]
bors merge

**Pull request merged**